smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaThrottledLatestProcessedCommit.java

```diff
@@ -105,11 +104,50 @@ private OffsetStore getOffsetStore(TopicPartition topicPartition) {
      */
     @Override
     public void partitionsAssigned(Set<TopicPartition> partitions) {
-        offsetStores.clear();
+        stopFlushAndCheckHealthTimer();
+
+        // Remove all handled partitions that are not in the given list of partitions
+        for (TopicPartition partition : new HashSet<>(offsetStores.keySet())) {
+            if (!partitions.contains(partition)) {
+                offsetStores.remove(partition);
+            }
+        }
+
+        // We cannot commit the removed offsets as we are not assigned to this partition anymore.
+
+        if (!offsetStores.isEmpty()) {
+            startFlushAndCheckHealthTimer();
+        }
```

Comment:
If we are not adding partitions to the store in this method then we need to start the time if either that store or partitions aren't empty.

Reply:
Yes!

Reply:
Done!